src/app/views/query-runner/request/headers/RequestHeaders.tsx

```diff
@@ -63,76 +57,52 @@ class RequestHeaders extends Component<IRequestHeadersProps, any> {
 
       const newHeaders = [header, ...sampleHeaders];
 
-      this.setState({
-        headerName: '',
-        headerValue: '',
-      });
+      setHeaderName('');
+      setHeaderValue('');
+      setAnnouncedMessage(translateMessage('Request Header added'));
 
-      if (actions) {
-        const query = this.props.sampleQuery;
-        query.sampleHeaders = newHeaders;
-        actions.setSampleQuery(query);
-      }
+      const query = { ...sampleQuery };
+      query.sampleHeaders = newHeaders;
+      dispatch(queryInputActionCreators.setSampleQuery(query));
     }
   };
 
-  public render() {
-    // @ts-ignore
-    const { sampleQuery, intl: { messages } } = this.props;
-    const headers = sampleQuery.sampleHeaders;
-    const container: any = headerStyles().container;
-
-    return (
-      <div style={container}>
-        <div className='row'>
-          <div className='col-sm-5'>
-            <TextField className='header-input'
-              placeholder={messages.Key}
-              value={this.state.headerName}
-              onChange={(event, name) => this.handleOnHeaderNameChange(name)}
-            />
-          </div>
-          <div className='col-sm-5'>
-            <TextField
-              className='header-input'
-              placeholder={messages.Value}
-              value={this.state.headerValue}
-              onChange={(event, value) => this.handleOnHeaderValueChange(value)}
-            />
-          </div>
-          <div className='col-sm-2 col-md-2'>
-            <PrimaryButton
-              style={{ width: '100%' }}
-              onClick={() => this.handleOnHeaderAdd()}>
-              <FormattedMessage id='Add' />
-            </PrimaryButton>
-          </div>
+  return (
+    <div style={container}>
```

Comment:
Let's not pass inline styles because they become really hard to debug. Class names are easier to work with.